### ISSUE TITLE
Refactor Object/Data Table

### ISSUE THREAD

**Issue by adrianq** [EyeSeeTea/d2-ui-components#68]
New Object Table:
- [x] New props
- [x] Remove dependency with d2 list (allows function and object)
- [x] Smart Formatter
- Features:
   - [x] Filters
   - [x] Pager
   - [x] Details panel (Details Box)

New Data Table:
- [x] Hooks
- [x] Typescripts
- Features:
   - [x] Primary action (left button)
   - [x] Control + click
   - [x] Contextual action (name + icons)
   - [x] Single action vs bulk actions (contextual action)
   - [x] Sorting
   - [x] 3-dots at the end of the line
   - [x] Selection
   - [x] Selection in all pages
- [x] Change components to material v4 (or v3)
- [x] Remove sass? 

New Feature:
- [x] Add property for rows per page

------------------------------------------
New Data Table - Phase II:
- [ ] Test
- [ ] Layout settings (show/hide column as in user app)
- [x] Uncontrolled behavior
- [x] Styles

New Object Table - Phase II:
- [x] Typescript
- [ ] Test

**Comment by SferaDev** [comment]
Before the meeting of Tuesday I have prepared a proof of concept for the new Data Table. The main objective of the PoC is to provide a workable (very far from finished) component to test and approve the functional requirements.

It currently includes almost all the objectives of this issue (typescript, hooks, primary action, control-click, details box, sorting, etc...).

It would be awesome that @adrianq can try it out today, so during tomorrow's meeting we can discuss what to improve/change before building the final implementation.

Note that the code is not final and for simplicity I have joined everything in the same file, this will change during the development process.

The preview is available at: https://codesandbox.io/embed/new-data-table-template-mvju7?autoresize=1&fontsize=14&hidenavigation=1&view=preview
